framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -117,6 +116,55 @@ AutomaticMortarGeneration::buildNodeToElemMaps()
   }
 }
 
+std::vector<Point>
+AutomaticMortarGeneration::getNodalNormals(const Elem & secondary_elem) const
+{
+  std::vector<Point> nodal_normals(secondary_elem.n_nodes());
+  for (const auto n : make_range(secondary_elem.n_nodes()))
+    nodal_normals[n] = secondary_node_to_nodal_normal.at(secondary_elem.node_ptr(n));
```

Comment:
Here, the normals belong to lower-dimensional elements, ie. `secondary_elem` is always a one-dimensional element?

Reply:
Correct...this should only ever be called with a lower dimensional element from the secondary face

Reply:
I should perhaps add an assertion that `secondary_elem.dim() == 1`

Reply:
But there is a way forward to do this for 2d secondary elements, right?